### PR TITLE
TELCODOCS-1508: Doc improvements from SME feedback

### PR DESCRIPTION
**Pull request by rohennes**
[TELCODOCS-1508](https://issues.redhat.com//browse/TELCODOCS-1508): SME requested doc improvements and some questions regarding functionality. 

Version(s):
4.12+

Issue:
https://issues.redhat.com/browse/TELCODOCS-1508

Link to docs preview:
https://70756--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/multiple_networks/assigning-a-secondary-network-to-a-vrf

QE review:
- [x] QE has approved this change.

Additional information:
More detailed SME feedback here: https://docs.google.com/document/d/1FSxqVwFT7uYT7obhjuZpMryuu5eVd5FekUHZZ67V2MM/edit?usp=sharing

### CODE REVIEW

**Comment by kalexand-rh** [comment]
4.11 is now [EOL](https://access.redhat.com/support/policy/updates/openshift), and we are no longer updating it. Please set the versions to 4.12+.

File: modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc [hunk 1]
```diff
@@ -43,33 +30,33 @@ apiVersion: operator.openshift.io/v1
...
-    }'
+    - name: test-network-1
+      namespace: additional-network-1
+      type: Raw
+      rawCNIConfig: '{
+        "cniVersion": "0.3.1",
+        "name": "macvlan-vrf",
+        "plugins": [  <1>
+        {
+          "type": "macvlan",  <2>
```
Comment (evgenLevin) [inline modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc:43]: A bit confused. The description here specifies that the type must be set to "vrf". I believe it should be pointed to the second item in the list.

**Review by evgenLevin — COMMENTED** [review]
A minor comment unrelated to this PR.

File: networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc [hunk 2]
```diff
@@ -6,4 +6,23 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
+As a cluster administrator, you can configure an additional network for a virtual routing and forwarding (VRF) domain by using the CNI VRF plugin. The virtual network created by this plugin is associated with a physical interface that you specify.
```
Comment (jeana-redhat) [inline networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc:9]: Spell out CNI on first appearance?

Comment (jeana-redhat) [reply networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc:9]:
```suggestion
As a cluster administrator, you can configure an additional network for a virtual routing and forwarding (VRF) domain by using the CNI VRF plugin. The virtual network that this plugin creates is associated with the physical interface that you specify. 
```

File: networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc [hunk 3]
```diff
@@ -6,4 +6,23 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
+As a cluster administrator, you can configure an additional network for a virtual routing and forwarding (VRF) domain by using the CNI VRF plugin. The virtual network created by this plugin is associated with a physical interface that you specify. 
+
+Using a secondary network with a VRF instance has the following advantages:
+
+Workload isolation:: Isolate workload traffic by configuring the additional network's VRF instance.
```
Comment (jeana-redhat) [inline networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc:13]:
```suggestion
Workload isolation:: Isolate workload traffic by configuring a VRF instance for the additional network.
```

File: networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc [hunk 4]
```diff
@@ -6,4 +6,23 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
+As a cluster administrator, you can configure an additional network for a virtual routing and forwarding (VRF) domain by using the CNI VRF plugin. The virtual network created by this plugin is associated with a physical interface that you specify. 
+
+Using a secondary network with a VRF instance has the following advantages:
+
+Workload isolation:: Isolate workload traffic by configuring the additional network's VRF instance. 
+Improved security:: Enable improved security through isolated network paths in the VRF domain.
+Multi-tenancy support:: Support multi-tenancy through network segmentation with a unique routing table in the VRF domain for each tenant.
```
Comment (jeana-redhat) [inline networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc:15]: ISG says not to use a hyphen here, but I imagine that's a common word in this content area. Would leave it alone for consistency, but consider updating the overall content to use "multitenancy" as a separate effort.

File: networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc [hunk 5]
```diff
@@ -6,4 +6,23 @@ include::_attributes/common-attributes.adoc[]
...
+
+[NOTE]
+====
+Applications that use VRFs need to bind to a specific device. The common usage is to use the `SO_BINDTODEVICE` option for a socket. `SO_BINDTODEVICE` binds the socket to a device that is specified in the passed interface name, for example, `eth1`. To use `SO_BINDTODEVICE`, the application must have `CAP_NET_RAW` capabilities.
```
Comment (jeana-redhat) [inline networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc:19]:
```suggestion
Applications that use VRFs must bind to a specific device. The common usage is to use the `SO_BINDTODEVICE` option for a socket. The `SO_BINDTODEVICE` option binds the socket to the device that is specified in the passed interface name, for example, `eth1`. To use the `SO_BINDTODEVICE` option, the application must have `CAP_NET_RAW` capabilities.
```

File: modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc
Comment (jeana-redhat) [inline modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc:1]: You could omit line 16 in here ("To create an additional network attachment with the CNI VRF plugin, perform the following procedure.")

File: modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc [hunk 6]
```diff
@@ -43,33 +30,33 @@ apiVersion: operator.openshift.io/v1
...
-        "table": 1001   <4>
-      }]
-    }'
+    - name: test-network-1
+      namespace: additional-network-1
+      type: Raw
+      rawCNIConfig: '{
+        "cniVersion": "0.3.1",
+        "name": "macvlan-vrf",
+        "plugins": [  <1>
```
Comment (jeana-redhat) [inline modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc:41]:
```suggestion
        "plugins": [ # <1>
```

File: modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc [hunk 7]
```diff
@@ -43,33 +30,33 @@ apiVersion: operator.openshift.io/v1
...
+              {
+                  "address": "191.168.1.23/24"
+              }
+              ]
+          }
+        },
+        {
+          "type": "vrf", <2>
+          "vrfname": "vrf-1",  <3>
+          "table": 1001   <4>
```
Comment (jeana-redhat) [inline modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc:57]:
```suggestion
          "type": "vrf", # <2>
          "vrfname": "vrf-1", # <3>
          "table": 1001 # <4>
```

File: modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc [hunk 8]
```diff
@@ -107,12 +94,48 @@ additional-network-1       14m
 There might be a delay before the CNO creates the CR.
 ====
 
-.Verifying that the additional VRF network attachment is successful
+.Verification
+
+To verify that the additional network is deployed and associated with the VRF, complete the following steps:
```
Comment (jeana-redhat) [inline modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc:100]: Should omit this
```suggestion
```

File: modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc [hunk 9]
```diff
@@ -107,12 +94,48 @@ additional-network-1       14m
...
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+ name: pod-additional-net
+ annotations:
+   k8s.v1.cni.cncf.io/networks: '[
+       {
+               "name": "test-network-1" <1>
```
Comment (jeana-redhat) [inline modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc:115]:
```suggestion
               "name": "test-network-1" # <1>
```

File: modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc [hunk 10]
```diff
@@ -107,12 +94,48 @@ additional-network-1       14m
...
++
+.Example output
+[source,terminal]
+----
+pod/test-pod created
+----
 
-. Create a network that uses the VRF CNI.
-. Assign the network to a pod.
 . Verify that the pod network attachment is connected to the VRF additional network. Remote shell into the pod and run the following command:
```
Comment (jeana-redhat) [inline modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc:139]: Is there a way to rephrase this to avoid using "remote shell" as a verbal phrase?

File: modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc [hunk 11]
```diff
@@ -121,22 +144,20 @@ $ ip vrf show
...
-+
 [source,terminal]
 ----
 Name              Table
 -----------------------
-red                 10
+vrf-1             1001
 ----
-. Confirm the VRF interface is master of the secondary interface:
+. Confirm that the VRF interface is master of the additional interface:
```
Comment (jeana-redhat) [inline modules/cnf-assigning-a-secondary-network-to-a-vrf.adoc:153]: This needs to be rephrased to either remove the term "master" or (if that is not possible), reframe it as a part of the code. For machines, we sometimes must refer to the control plane nodes as "nodes that have the `master` role", since `role: master` is in the spec. I don't know what the equivalent is for this case, but (if you can't omit the term entirely) find out how to make it describe its technical aspect specifically.

**Review by jeana-redhat — COMMENTED** [review]
This looks great! A few things that you should address, and otherwise some suggestions.

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

File: networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc (hunk 2 above)
Comment (rohennes) [reply networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc:9]: Regarding CNI, I am tempted to leave this alone as it's part of the plugin name? Container Network Interface (CNI) VRF plugin is a bit confusing?

Comment (jeana-redhat) [reply networking/multiple_networks/assigning-a-secondary-network-to-a-vrf.adoc]: oh yeah if the name is the abbreviated version that makes sense to me so the user will find it :+1:
